var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -91,38 +101,53 @@ def install(self, spec, prefix):
             options.append('--enable-openmp')
 
         if '+scalapack' in spec:
-            options.append('--with-scalapack=yes')
+            scalapack_option = 'intel' if '^intel-mkl' in spec else 'yes'
+            options.append('--with-scalapack={0}'.format(scalapack_option))
 
         if '+elpa' in spec:
-            if spec.satisfies('@:5.4.0'):
-                options.append('--with-elpa=yes')
-            else:
-                options.extend([
-                    '--with-elpa-include={0}'.format(
-                        join_path(spec['elpa'].prefix,
-                                  'include',
-                                  'elpa-{0}'.format(str(spec['elpa'].version)),
-                                  'modules')
-                    ),
-                    '--with-elpa-lib={0}'.format(spec['elpa'].libs.joined())
-                ])
+
+            # Spec for elpa
+            elpa = spec['elpa']
+
+            # Find where the Fortran module resides
+            elpa_module = find(elpa.prefix, 'elpa.mod')
+
+            # Compute the include directory from there: versions
+            # of espresso prior to 6.1 requires -I in front of the directory
+            elpa_include = '' if '@6.1:' in spec else '-I'
+            elpa_include += os.path.dirname(elpa_module[0])
+
+            options.extend([
+                '--with-elpa-include={0}'.format(elpa_include),
+                '--with-elpa-lib={0}'.format(elpa.libs[0])
+            ])
+
+        if '+hdf5' in spec:
+            options.append('--with-hdf5={0}'.format(spec['hdf5'].prefix))
```

Comment:
I need to check that. As I said this is a hand-crafted `configure`, so I am not really sure if it does things the way autotools would.